sceptre/hooks/cmd.py

```diff
@@ -13,16 +13,33 @@ def __init__(self, *args, **kwargs):
 
     def run(self):
         """
-        Runs the argument string in a subprocess.
+        Executes a command through the shell.
 
-        :raises: sceptre.exceptions.InvalidTaskArgumentTypeException
+        See hooks documentation for details.
+
+        :raises: sceptre.exceptions.InvalidHookArgumentTypeError
         :raises: subprocess.CalledProcessError
         """
         envs = self.stack.connection_manager.create_session_environment_variables()
-        try:
-            subprocess.check_call(self.argument, shell=True, env=envs)
-        except TypeError:
+
+        if isinstance(self.argument, str) and self.argument != "":
+            command_to_run = self.argument
+            shell = None
+
+        elif (
+            isinstance(self.argument, dict)
+            and set(self.argument) == {"run", "shell"}
+            and isinstance(self.argument["run"], str)
+            and isinstance(self.argument["shell"], str)
+        ):
+            command_to_run = self.argument["run"]
+            shell = self.argument["shell"]
```

Comment:
i didn't even consider windows.  What happens if users attempt to use a windows shell on linux or the other way around?  should we bring back the try/catch or add some error handling around `shell` inputs as well?

Reply:
I'm not against adding a try/catch around the shell invocation. I just don't know what would be useful to catch yet. It might be useful later to wrap the stock Python exception with a Sceptre exception for nicer presentation of the errors, but I had considered that out of scope for now.

I don't think it's our business to know which shells are available in the environment. If the user tells Sceptre to use a shell that isn't available, then Sceptre should fail. I'll add a test and documentation to show what that failure looks like. This is useful to know how it behaves when I do something silly like type `bsah` instead of `bash`.

(PowerShell ships with Windows, but it's no longer just a shell for Windows. I [installed it on Ubuntu](https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-linux?view=powershell-7.3) to test the `pwsh` part of the documentation! Similarly, I believe [Windows for Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install) users can use the Bash shell.)

Reply:
hmm, `FileNotFoundError` for invalid shell name is not very helpful.  would you be able to provide a more helpful message?

Reply:
The full error message is `FileNotFoundError: [Errno 2]` No such file or directory: '/bin/bsah'`. It does print the bad shell name. But we could improve the message with a Sceptre error that tells the user to check the shell input.

Since we've started talking about custom error handling, it would be great to suppress the stack trace for these kinds of errors that we know how to handle. Currently Sceptre will print about 50 lines of unhelpful stack trace between the first useful error message and the last useful error message. I'll see what I can do about that.

Reply:
ohh, the full message looks good to me.  It's up to you but I don't think you need to add supression of the stack trace on this PR.  i'm ok if you want to do it as a separate change.

Reply:
Thanks. I agree about using another PR. The stack trace suppression happens elsewhere in the code. I don't want to make this PR any wider.